samples/sample_common/src/vaapi_utils_drm.cpp

```diff
@@ -22,216 +22,91 @@ or https://software.intel.com/en-us/media-client-solutions-support.
+const char* MFX_DRI_NODE_CARD = "card";
 
-struct mfx_disp_adapters
+int get_drm_driver_name(int fd, char *name, int name_size)
 {
-    mfxU32 vendor_id;
-    mfxU32 device_id;
-};
-
-static int mfx_dir_filter(const struct dirent* dir_ent)
-{
-    if (!dir_ent) return 0;
-    if (!strcmp(dir_ent->d_name, ".")) return 0;
-    if (!strcmp(dir_ent->d_name, "..")) return 0;
-    return 1;
+    drm_version_t version = {};
+    version.name_len = name_size;
+    version.name = name;
+    return ioctl(fd, DRM_IOWR(0, drm_version), &version);
 }
 
-typedef int (*fsort)(const struct dirent**, const struct dirent**);
-
-static mfxU32 mfx_init_adapters(struct mfx_disp_adapters** p_adapters)
+int open_intel_adapter(int type)
 {
-    mfxU32 adapters_num = 0;
-    int i = 0;
-    struct mfx_disp_adapters* adapters = NULL;
-    struct dirent** dir_entries = NULL;
-    int entries_num = scandir(MFX_PCI_DIR, &dir_entries, mfx_dir_filter, (fsort)alphasort);
-
-    char file_name[300] = {};
-    char str[16] = {0};
-    FILE* file = NULL;
-
-    for (i = 0; i < entries_num; ++i)
-    {
-        long int class_id = 0, vendor_id = 0, device_id = 0;
-
-        if (!dir_entries[i])
-            continue;
-
-        // obtaining device class id
-        snprintf(file_name, sizeof(file_name)/sizeof(file_name[0]), "%s/%s/%s", MFX_PCI_DIR, dir_entries[i]->d_name, "class");
-        file = fopen(file_name, "r");
-        if (file)
-        {
-            if (fgets(str, sizeof(str), file))
-            {
-                class_id = strtol(str, NULL, 16);
-            }
-            fclose(file);
-
-            if (MFX_PCI_DISPLAY_CONTROLLER_CLASS == (class_id >> 16))
-            {
-                // obtaining device vendor id
-                snprintf(file_name, sizeof(file_name)/sizeof(file_name[0]), "%s/%s/%s", MFX_PCI_DIR, dir_entries[i]->d_name, "vendor");
-                file = fopen(file_name, "r");
-                if (file)
-                {
-                    if (fgets(str, sizeof(str), file))
-                    {
-                        vendor_id = strtol(str, NULL, 16);
-                    }
-                    fclose(file);
-                }
-                // obtaining device id
-                snprintf(file_name, sizeof(file_name)/sizeof(file_name[0]), "%s/%s/%s", MFX_PCI_DIR, dir_entries[i]->d_name, "device");
-                file = fopen(file_name, "r");
-                if (file)
-                {
-                    if (fgets(str, sizeof(str), file))
-                    {
-                        device_id = strtol(str, NULL, 16);
-                    }
-                    fclose(file);
-                }
-                // adding valid adapter to the list
-                if (vendor_id && device_id)
-                {
-                    struct mfx_disp_adapters* tmp_adapters = NULL;
+    std::string adapterPath = MFX_DRI_PATH;
+    char driverName[MFX_DRM_DRIVER_NAME_LEN + 1] = {};
+    mfxU32 nodeIndex = MFX_DRI_RENDER_START_INDEX;
```

Comment:
Can you leave nodeIndex either not initialized or initialized to 0? Move MFX_DRI_RENDER_START_INDEX to render start section.

Reply:
fixed